components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java

```diff
@@ -966,13 +957,53 @@ private static String retrieveStateForErrorURL(OAuth2Parameters oAuth2Parameters
         return state;
     }
 
+    /**
+     * Return updated redirect URL.
+     *
+     * @param request       HttpServletRequest
+     * @param redirectUri   Redirect Uri
+     * @param errorCode     Error Code
+     * @param errorMessage  Message of the error
+     * @param state         State from the request
+     * @param appName       Application Name
+     * @return Updated Redirect URL
+     */
+    private static String getUpdatedRedirectURL(HttpServletRequest request, String redirectUri, String errorCode,
+                                                String errorMessage, String state, String appName) {
+
+        String updatedRedirectUri = redirectUri;
+        try {
+            OAuthProblemException ex = OAuthProblemException.error(errorCode).description(errorMessage);
+            if (OAuth2Util.isImplicitResponseType(request.getParameter(OAuthConstants.OAuth20Params.RESPONSE_TYPE))
+                    || OAuth2Util.isHybridResponseType(request.getParameter(OAuthConstants.OAuth20Params.
+                    RESPONSE_TYPE))) {
+                updatedRedirectUri = OAuthASResponse.errorResponse(HttpServletResponse.SC_FOUND)
+                        .error(ex).location(redirectUri).setState(state).setParam(OAuth.OAUTH_ACCESS_TOKEN, null)
+                        .buildQueryMessage().getLocationUri();
+            } else {
+                updatedRedirectUri = OAuthASResponse.errorResponse(HttpServletResponse.SC_FOUND)
+                        .error(ex).location(redirectUri).setState(state).buildQueryMessage().getLocationUri();
+            }
+
+        } catch (OAuthSystemException e) {
+            log.error("Server error occurred while building error redirect url for application: " + appName, e);
+        }
+        return updatedRedirectUri;
+    }
+
     /**
      * Method to retrieve the <AllowAdditionalParamsFromErrorUrl> config from the OAuth Configuration.
      * @return Retrieved config (true or false)
      */
     private static boolean isAllowAdditionalParamsFromErrorUrlEnabled() {
 
-        return Boolean.parseBoolean(IdentityUtil.getProperty(ALLOW_ADDITIONAL_PARAMS_FROM_ERROR_URL));
+        String isAllowAdditionalParamsEnabled = IdentityUtil.getProperty(ALLOW_ADDITIONAL_PARAMS_FROM_ERROR_URL);
+
+        if (StringUtils.isNotBlank(isAllowAdditionalParamsEnabled)) {
```

Comment:
Shouldn't this be the other way in the master?

ie. by default we should not allow additional params in the error URL as it violates the spec

Reply:
Made isAllowAdditionalParamsEnabled be false by default. Thank you for pointing that out.